### PR TITLE
Add redux functionality

### PR DESCRIPTION
**Pull request by aliatayee**
## Redux Functionality
In this milestone I have developed the following parts:
- [ ] Update reducers to make it immutable
- [ ] Add ```Provider```
- [ ] Add form functionality to push the new book to the array using ```dispatch```
- [ ] Add remove functionality to remove book using ```dispatch```
- [ ] Add category button functionality to consume state and dispatch the ```checkStatus``` action
- [ ] Update README.md file

### CODE REVIEW

File: src/components/Form.js [hunk 1]
```diff
@@ -1,29 +1,52 @@
...
-    <div className="md:flex">
-      <div className="md:w-1/2 sm:w-full md:pr-4">
-        <input
-          type="email"
-          className="form-input mt-3 block w-full rounded-md border-gray-300 shadow-sm py-2 rounded"
-          placeholder="Book Title"
-        />
+  const handleForm = (e) => {
+    e.preventDefault();
+    const [title, author] = e.target.elements;
```
Comment (TedLivist) [inline src/components/Form.js]: - [ ] Your inputs should be controlled and so their values should be set by local states, not direct input values. Kindly set your title and author to be states. This state is what should be handled by the submit action.

- [ ] Each of the HTML inputs should have methods triggered by the `onChange` event, which sets the states for the title and author.

- [ ] Next the values of the inputs can be assigned using the `value` HTML attribute.

  You can take a look at [this article on how to do that](https://dmitripavlutin.com/controlled-inputs-using-react-hooks/) or the [React documentation for Forms](https://reactjs.org/docs/forms.html)

File: README.md [hunk 2]
```diff
@@ -16,13 +16,13 @@ To setup and install this sample Leaderboard project, follow the below steps:
 - Clone this project by the command:
```
Comment (TedLivist) [inline README.md:16]: - [ ] At the top of your documentation, after the title, kindly include a sentence or paragraph describing the project as we have on [the Microverse template](https://github.com/microverseinc/readme-template).

  readme-ff

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @aliatayee,

Good job so far!
✔️ Books can be removed
✔️ App is wrapped by the Provider
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
### **Hi @aliatayee,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**